pkg/controller/scansettingbinding/scansettingbinding_controller.go

```diff
@@ -0,0 +1,421 @@
+
+	tailoredProfile *unstructured.Unstructured
+	profile         *unstructured.Unstructured
+	profileBundle   *unstructured.Unstructured
+}
+
+func profileReferenceToScan(reference *profileReference) (*compliancev1alpha1.ComplianceScanSpecWrapper, error) {
+	var err error
+	//var found bool
+
+	scan := compliancev1alpha1.ComplianceScanSpecWrapper{
+		ComplianceScanSpec: compliancev1alpha1.ComplianceScanSpec{},
+		Name:               reference.name,
+	}
+
+	err = fillContentData(reference.profileBundle, &scan)
+	if err != nil {
+		return nil, err
+	}
+
+	if reference.tailoredProfile != nil {
+		err = fillTailoredProfileData(reference.tailoredProfile, &scan)
+		if err != nil {
+			return nil, err
+		}
+	} else if reference.profile != nil {
+		err = fillProfileData(reference.profile, &scan)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("neither profile nor tailoredProfile are known")
+	}
+
+	return &scan, nil
+}
+
+func isGvk(obj *unstructured.Unstructured, expectGvk *schema.GroupVersionKind) error {
+	objGvk := obj.GetObjectKind().GroupVersionKind()
+
+	if objGvk.Kind != expectGvk.Kind {
+		return fmt.Errorf("expected Kind %s, received %s", expectGvk.Kind, objGvk.Kind)
+	}
+
+	if objGvk.Version != expectGvk.Version {
+		return fmt.Errorf("expected Version %s, received %s", expectGvk.Version, objGvk.Version)
+	}
+
+	if objGvk.Group != expectGvk.Group {
+		return fmt.Errorf("expected Group %s, received %s", expectGvk.Group, objGvk.Group)
+	}
+
+	return nil
+}
+
+func fillContentData(bundle *unstructured.Unstructured, scan *compliancev1alpha1.ComplianceScanSpecWrapper) error {
+	// TODO: if we even support multiple versions, add an array of gvk:handler_fn tuples
+	expGvk := schema.GroupVersionKind{
+		Group:   compliancev1alpha1.SchemeGroupVersion.Group,
+		Version: compliancev1alpha1.SchemeGroupVersion.Version,
+		Kind:    "ProfileBundle",
+	}
+
+	if err := isGvk(bundle, &expGvk); err != nil {
+		return err
+	}
+
+	v1alphaBundle := compliancev1alpha1.ProfileBundle{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(bundle.Object, &v1alphaBundle)
+	if err != nil {
+		return err
+	}
+
+	scan.Content = v1alphaBundle.Spec.ContentFile
+	scan.ContentImage = v1alphaBundle.Spec.ContentImage
+	return nil
+}
+
+func fillTailoredProfileData(tp *unstructured.Unstructured, scan *compliancev1alpha1.ComplianceScanSpecWrapper) error {
+	expGvk := schema.GroupVersionKind{
+		Group:   compliancev1alpha1.SchemeGroupVersion.Group,
+		Version: compliancev1alpha1.SchemeGroupVersion.Version,
+		Kind:    "TailoredProfile",
+	}
+
+	if err := isGvk(tp, &expGvk); err != nil {
+		return err
+	}
+
+	v1alphaTp := compliancev1alpha1.TailoredProfile{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(tp.Object, &v1alphaTp)
+	if err != nil {
+		return err
+	}
+
+	scan.Profile = v1alphaTp.Status.ID
+	if v1alphaTp.Status.OutputRef.Name != "" {
+		// FIXME: OutputRef also has a namespace, but tailorringCofnigMapRef not?
```

Comment:
Right, that's because the TailoringconfigMapRef that's used in the ComplianceScan is required to be in the same namespace as the scan. It was modeled similar to a pod, where the reference to the configmap for a pod is also without namespace since the namespace is assumed.

However, when outputting a reference, I wanted to be explicit and mention the namespace too since that would allow for doing extra checks on the object, and surface errors (did we output the configmap on the operator namespace or the watch namespace?).

Feel free to change it (add remove a namespace parameter) as you see fit.

Reply:
I think so far it is OK, I just put the FIXME in the code to remember not to break any of the flows we have with RHACM.